tools/python/python_bootstrap_template.txt

```diff
@@ -354,11 +355,20 @@ def ExecuteFile(python_program, main_filename, args, env, module_space,
     ret_code = _RunForCoverage(python_program, main_filename, args, env,
                                coverage_entrypoint, workspace)
   else:
-    ret_code = subprocess.call(
-      [python_program, main_filename] + args,
-      env=env,
-      cwd=workspace
-    )
+    # Temporarily ignoring SIGINT to not interfere with python_program's own
+    # interrupt handling.
+    def noop(sig, frame):
+      pass
```

Comment:
Isn't it necessary to forward the SIGINT to the wrapped process? Or does this happen automatically somehow?

Reply:
My understanding is that `SIGINT` is automatically sent to all processes within a process group.
And since `python_program` is spawned with `subprocess.call` **without** `start_new_session=True`, it will get the same pgid.

Please correct me if I'm wrong, or missing a significant edge case.

This can be observed with this program:

<details><summary>Click to expand</summary>
<p>

```py
import os
import signal
import subprocess as sp
import sys
import time

pid = os.getpid()
pgid = os.getpgid(pid)

is_parent_process = len(sys.argv) == 1

print("parent" if is_parent_process else "child")
print(f"  {pid=}")
print(f"  {pgid=}")

def noop(sig, frame):
  pass

try:
  if is_parent_process:
    # signal.signal(signal.SIGINT, noop)
    sp.call(
      [sys.executable, __file__, "child"],
      # start_new_session=True,
    )
  else:
    print("press CTRL+C now")
    time.sleep(1000)
except:
  exc_type = sys.exc_info()[0]
  print("Exception caught", pid, "of type:", exc_type)

```

If you run it, you'll see two processes spawned:
```
parent
  pid=89304
  pgid=89304
child
  pid=89305
  pgid=89304
press CTRL+C now
```
and if you interrupt it, you will see that both of them receve `KeyboardInterrupt`:
```
…
press CTRL+C now
^CException caught 89305 of type: <class 'KeyboardInterrupt'>
Exception caught 89304 of type: <class 'KeyboardInterrupt'>
```

---

Now if you uncomment `signal.signal(signal.SIGINT, noop)` and repeat the steps, you will see that the parent sinks the interrupt into `noop` but the child still receives it as before:
```
parent
  pid=89584
  pgid=89584
child
  pid=89585
  pgid=89584
press CTRL+C now
^CException caught 89585 of type: <class 'KeyboardInterrupt'>
```

---

If you uncomment `start_new_session=True,`,
📝  **and make sure to comment `signal.signal(signal.SIGINT, noop)` back out again**,
then you'll see that the child is spawned with a different `pgid` this time, and no longer receives keyboard interrupts at all:
```
parent
  pid=89614
  pgid=89614
child
  pid=89615
  pgid=89615
press CTRL+C now
^CException caught 89614 of type: <class 'KeyboardInterrupt'>
```

</p>
</details>

Reply:
Great, I didn't know that. Thanks for the detailed example! It could be worth adding a comment with a short explanation so that future readers of this part of the code don't see the same bug that isn't there.